cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -183,6 +183,70 @@ def _wait_for_cfy_node_to_start_serving(cfy, timeout=15):
     raise TimeoutException
 
 
+def test_workflow_resume_manager_failover(minimal_cluster, cfy):
+    broker, db, mgr1, mgr2 = minimal_cluster
+    mgr1.use(cert_path=mgr1.local_ca)
+    plugin_name = 'waiting'
+    plugin_yaml = '{}-blueprint.yaml'.format(plugin_name)
+    plugin_wagon = 'testplugin-0.0.0-py27-none-any.wgn'
+
+    # upload a mock plugin which waits 1 minute, deploy and execute
+    tests_root = path.dirname(path.dirname(path.dirname(
+        path.realpath(__file__))))
+    blueprints_dir = path.join(tests_root, 'resources', 'blueprints', 'mocks')
+    cfy.blueprints.upload(path.join(blueprints_dir, plugin_yaml),
+                          '-b', plugin_name)
+    cfy.plugins.upload(path.join(blueprints_dir, plugin_wagon), '-y',
+                       path.join(blueprints_dir, plugin_yaml))
+    cfy.deployments.create(plugin_name, '-b', plugin_name)
+    _wait_for_deployment_creation(cfy, plugin_name)
+    execution_start_time = time.time()
+    mgr1.client.executions.start(plugin_name, 'install')
+
+    # check which management worker handles the execution
+    exec_node_id = json.loads(
+        mgr1.run_command('cfy node-instances list --json').lstrip(
+            '\r\n\033[0m').rstrip('\r\n\033[0m]'))['id']
+    time.sleep(3)   # wait for mgmtworker to get the execution
+    executing_manager = None
+    other_manager = None
+    for manager in [mgr1, mgr2]:
+        try:
+            manager.run_command(
+                'cat /var/log/cloudify/mgmtworker/mgmtworker.log '
+                '| grep {}'.format(exec_node_id))
+        except Exception:
+            continue
+        executing_manager = manager
+        other_manager = ({mgr1, mgr2} - {executing_manager}).pop()
+
+    # kill the first manager and  wait for execution to finish
+    with executing_manager.ssh() as manager_ssh:
+        manager_ssh.run('cfy_manager stop')
+    manager_failover_time = time.time()
+    assert manager_failover_time - execution_start_time < 60
+    time.sleep(60)
+    other_manager.use(cert_path=other_manager.local_ca)
+
+    # verify on the second manager that the execution completed successfully
+    executions = json.loads(cfy.executions.list('--json').stdout)
+    installs = [execution for execution in executions
+                if execution['workflow_id'] == 'install']
+    assert len(installs) == 1
+    assert installs[0]['status'] == 'completed'
+
+
+def _wait_for_deployment_creation(cfy, deployment_id, timeout=15):
```

Comment:
unfortunate, we have something like this function in about 10000 places, we should make an official real one already